tensorflow/lite/kernels/split.cc

```diff
@@ -111,24 +114,22 @@ TfLiteStatus Eval(TfLiteContext* context, TfLiteNode* node) {
     axis_value += NumDimensions(op_context.input);
   }
 
+  TF_LITE_ENSURE(context, axis_value >= 0);
+  TF_LITE_ENSURE(context, axis_value < NumDimensions(op_context.input));
+
   // TODO(ahentz): Our usage of VectorOfTensors could be optimized by
   // calculating it in Prepare, unless we defer shape calculation.
   // TODO(ahentz): We can improve the optimized_ops version to handle other
   // cases too.
-#define TF_LITE_SPLIT(scalar)                                         \
-  VectorOfTensors<scalar> all_outputs(*context, *node->outputs);      \
-  tflite::SplitParams op_params;                                      \
-  op_params.num_split = NumOutputs(node);                             \
-  op_params.axis = axis_value;                                        \
-  if (axis_value == 0) {                                              \
-    optimized_ops::Split(op_params, GetTensorShape(op_context.input), \
-                         GetTensorData<scalar>(op_context.input),     \
-                         all_outputs.shapes(), all_outputs.data());   \
-  } else {                                                            \
-    reference_ops::Split(op_params, GetTensorShape(op_context.input), \
-                         GetTensorData<scalar>(op_context.input),     \
-                         all_outputs.shapes(), all_outputs.data());   \
-  }
+#define TF_LITE_SPLIT(scalar)                                       \
+  VectorOfTensors<scalar> all_outputs(*context, *node->outputs);    \
+  tflite::SplitParams op_params;                                    \
+  op_params.num_split = NumOutputs(node);                           \
+  op_params.axis = axis_value;                                      \
+  reference_ops::Split(op_params, GetTensorShape(op_context.input), \
```

Comment:
It seems that optimized_ops and reference_ops share the same Split implementation(https://github.com/tensorflow/tensorflow/blob/fe9d5075d4c71e701f1acbe2cc267b1165d8cedd/tensorflow/lite/kernels/internal/optimized/optimized_ops.h#L88). I'm not sure why we call them separately here, Jared, is there any background knowledge that I might not be aware of?

Reply:
I think it's fine to just use reference_ops::Split now.